pkgs/development/python-modules/tatsu/default.nix

```
@@ -0,0 +1,46 @@
+{ stdenv, buildPythonPackage, fetchFromGitHub, pythonOlder
+, colorama, mypy, pyyaml, regex
+, dataclasses, typing
+, pytestrunner, pytest-mypy
+}:
+
+buildPythonPackage rec {
+  pname = "TatSu";
+  version = "4.4.0";
+
+  src = fetchFromGitHub {
+    owner = "neogeny";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1jjd73yr3x56ij2ggxf6s62mf90i9v7wn3i0h67zxys55hlp2yh4";
+  };
+
+  # TODO: Need to fix errors like this one:
+  # ______________________ ERROR collecting test/ast_test.py _______________________
+  # import file mismatch:
+  # imported module 'test.ast_test' has this __file__ attribute:
+  #   /build/source/dist/tmpbuild/TatSu/test/ast_test.py
+  # which is not the same as the test file we want to collect:
+  #   /build/source/test/ast_test.py
+  # HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
+  doCheck = false;
+
+  propagatedBuildInputs = [ colorama mypy pyyaml regex ]
+    ++ stdenv.lib.optionals (pythonOlder "3.7") [ dataclasses ]
+    ++ stdenv.lib.optionals (pythonOlder "3.5") [ typing ]
+    ++ stdenv.lib.optional (!doCheck) pytestrunner; # TODO
+  checkInputs = [ pytestrunner pytest-mypy ];
```

pytestrunner is a setup_reqiures package, should go to nativeBuildInputs
```suggestion
  checkInputs = [ pytestrunner pytest-mypy ];
```